src/main/java/com/atlassian/db/replica/api/DualConnection.java

```diff
@@ -520,14 +555,28 @@ private Builder(
             return this;
         }
 
+        /**
+         * Puts this connection in compatibility mode with a previous version. Developers can use this method to
+         * roll out the new version of the library with a feature flag.
+         *
+         * It's best-effort, and there's no guarantee the library in compatibility mode will always behave
+         * the same way as the previous version of the library.
+         *
+         */
+        public DualConnection.Builder compatibleWithPreviousVersion() {
```

Comment:
lack of passing param force consumer to define `DualConnection.Builder` in two places, instead of just passing the boolean value from FeatureFlag. Is it intentional?  If yes why?
So much duplicated code, maybe we could simplify it?

Reply:
You don't have to define the builder in two places.
```
 final DualConnection.Builder builder = DualConnection.builder(
            connectionProvider,
            consistency
        );
        if(oldVersionFlag){
            builder. compatibleWithPreviousVersion()
        }
```
When we pass a parameter, we can expect 3 values (true, false, null). You have 3 values to handle instead of one. False has no sense as it's a default. I don't know how to handle null properly here.